### PR TITLE
Bump flux to v0.37.0

### PR DESCRIPTION
**Pull request by uvegla**
Towards: https://github.com/giantswarm/giantswarm/issues/24832

### CODE REVIEW

**Review by kubasobon — COMMENTED** [review]
:fearful: Why are `ImagePolicy`, `ImageRepository`, and `ImageUpdateAutomation` CRs gone?

**Comment by uvegla** [comment]
> 😨 Why are `ImagePolicy`, `ImageRepository`, and `ImageUpdateAutomation` CRs gone?

It is just v1alpha1 one version gone.

File: hack/patches/kustomize-controller-deploy-patch.yaml [hunk 1]
```diff
@@ -16,6 +16,6 @@
 - op: replace
   path: /spec/template/metadata/annotations
   value: '{{ include "podTemplateAnnotations.kustomizeController" . }}'
-- op: replace
-  path: /spec/template/metadata/labels
-  value: '{{ include "podTemplateLabels.kustomizeController" . }}'
+#- op: replace
```
Comment (uvegla) [inline hack/patches/kustomize-controller-deploy-patch.yaml:19]: This had to be removed, cos it does work. The `kubectl kustomize` command fails that you try to replace a map with a scalar value.

File: helm/flux-app/templates/install.yaml [hunk 2]
```diff
@@ -902,7 +927,12 @@ spec:
               path: /
               port: http
           resources:
-{{ include "resources.sourceController" . | indent 12 }}
```
Comment (uvegla) [inline helm/flux-app/templates/install.yaml:905]: I am not sure about these, I assume the last time these were added to the kustomizations then kuztomize was not rerun.

**Comment by uvegla** [comment]
I think it is ok now. 👍🏻

**Comment by uvegla** [comment]
@kubasobon Could you please take another look / review?

File: helm/flux-app/templates/install.yaml (hunk 2 above)
Comment (kubasobon) [reply helm/flux-app/templates/install.yaml:905]: It was a VPA fix by @mproffitt https://github.com/giantswarm/flux-app/pull/151/files

**Comment by uvegla** [comment]
The diff looks much better now 👍🏻

**Pull request merged**